### PR TITLE
Use epics-base as dependency

### LINKED ISSUE #19: Use epics-base instead of epicscorelibs
I'm wondering whether it would be possible to switch out the dependency of this package to `epics-base`, rather than `epicscorelibs`. I've read through #17, and see that there are concerns that the `epics-base` package is broken on windows.

If we can resolve the epics-on-windows issues, is it reasonable to switch out the dependency, so that we can try and converge on `epics-base` as the one-true source of EPICS? I'm not an epics expert, just trying to simplify the packaging of `pyepics`, `pcaspy`, and a few other things needed by the LIGO experiment.

cc @jrollins.

### PR DESCRIPTION
**Pull request by beenje**
Checklist
* [X] Used a [fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Fixes #19

- Replace epicscorelibs with epics-base
- A strange thing is that the sha256 of the archive from GitHub changed...

### CODE REVIEW

**Comment by hhslepicka** [comment]
I just noticed that all windows builds are broken. I’m not sure if it is related to the PR. I will take a look

**Comment by hhslepicka** [comment]
@conda-forge-admin, please rerender

**Comment by hhslepicka** [comment]
It just needed a rerender... now CI is happy again.

File: recipe/activate.bat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@echo off
+set "PYEPICS_LIBCA=%EPICS_BASE%\bin\%EPICS_HOST_ARCH%\ca.dll"
```
Comment (tacaswell) [inline recipe/activate.bat:2]: It is not clear to me that these envs will be defined? On the other hand I am not a windows user and am going on pattern matching so please disregard if tihs is OK.

**Comment by tacaswell** [comment]
> A strange thing is that the sha256 of the archive from GitHub changed...

This is a known issue with versioneer, see https://matplotlib.org/devel/release_guide.html#id3 for the explanation.

I have a slight preference for still applying the libca patch to remove the non-standard library discovery mechanism (so things work even if you don't activate the env), but if it is not working I'm also not super willing to debug that right now.

It is also not clear to me how the windows activate.bat gets templated (I could just be ignorant of how that works).

I am happy if this gets merged, but would like my comment about the patch and the activate scripts to be intentionally dismissed.

File: recipe/activate.bat (hunk 1 above)
Comment (beenje) [reply recipe/activate.bat:2]: `EPICS_BASE` and `EPICS_HOST_ARCH` variables are defined in the the epics-base activate script: https://github.com/conda-forge/epics-base-feedstock/blob/master/recipe/pre-build.py#L17
The pyepics_activate.bat script has to be run after epics-base_activate.bat, which is the case as conda execute activate scripts following alphabetical order.

Comment (tacaswell) [reply recipe/activate.bat:2]: fair, but that seems a bit brittle.  Might be better to eventually prefix these with integers, but no reason to hold this PR up over that.

**Comment by beenje** [comment]
> > A strange thing is that the sha256 of the archive from GitHub changed...
> 
> This is a known issue with versioneer, see https://matplotlib.org/devel/release_guide.html#id3 for the explanation.
> 

We should maybe switch to PyPI archive to avoid this issue? This is the default in conda-forge examples: https://pypi.io/packages/source/p/pyepics/pyepics-3.4.1.tar.gz

> I have a slight preference for still applying the libca patch to remove the non-standard library discovery mechanism (so things work even if you don't activate the env), but if it is not working I'm also not super willing to debug that right now.
> 

What was exactly the purpose of the patch?

My preference would be to not patch the original package if not needed.

Note that for this package relies on activate scripts to define PYEPICS_LIBCA. If you don't activate the env, you'd have to set the variable manually.

> It is also not clear to me how the windows activate.bat gets templated (I could just be ignorant of how that works).
> 
As explained above, it works exactly like on linux/OSX. It relies on variables set by epics-base activation script.

> I am happy if this gets merged, but would like my comment about the patch and the activate scripts to be intentionally dismissed.

**Comment by hhslepicka** [comment]
The approach for the activate/deactivate is correct otherwise the test command would fail to find libca and raise an exception.

About removing the non-standard library discovery via the libca patch I honestly think that it will not make a difference. If you have `PYEPICS_LIBCA` defined it will find it otherwise the fallback approach will also fail as it was installed with `NOLIBCA=1`.

File: recipe/activate.bat (hunk 1 above)
Comment (hhslepicka) [reply recipe/activate.bat:2]: I will go ahead and mark this as resolved as the tests are passing and the current solution works. As @tacaswell mentioned we can address the prefix via another PR.

**Comment by tacaswell** [comment]
> What was exactly the purpose of the patch?

If we are using epics-base as the source of libca and friends, the patch should make them be found without having to set  PYEPICS_LIBCA (because the sos get dropped in a place that is in the standard search path).

**Comment by tacaswell** [comment]
Well, epicscorelibs is now broken https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=158806&view=logs&j=1bf226d3-0e2f-52d8-fa93-7d9e633347b3&t=ec8d466d-e3b4-5115-4f06-222398f91e6c&l=850 so I'm going to merge this and come back to removing the need for the ENV later.

**Comment by tacaswell** [comment]
oh, nvm, I don't have merge rights on this feedstock.

**Comment by hhslepicka** [comment]
The issue was not with epicscorelibs. It was in fact the bluesky feedstock missing `msgpack` and `msgpack-numpy` dependencies. But in any case, it was good to merge this PR in.

**Pull request merged**